aiida_quantumespresso/workflows/pdos.py

```diff
@@ -155,45 +174,49 @@ def clean_workchain_calcs(workchain):
 ProjwfcCalculation = plugins.CalculationFactory('quantumespresso.projwfc')
 
 
-class PdosWorkChain(engine.WorkChain):
+class PdosWorkChain(ProtocolMixin, WorkChain):
     """A WorkChain to compute Total & Partial Density of States of a structure, using Quantum Espresso."""
 
     @classmethod
     def define(cls, spec):
         # type: (ProcessSpec) -> None
         # yapf: disable
         """Define the process specification."""
-        super(PdosWorkChain, cls).define(spec)
+        super().define(spec)
+        spec.input('structure')
 
-        # Base SCF & NSCF inputs
         spec.expose_inputs(
-            PwBaseWorkChain, namespace='base',
-            exclude=('clean_workdir', 'pw.parent_folder'),
+            PwBaseWorkChain,
+            namespace='base_scf',
+            exclude=('clean_workdir', 'pw.structure', 'pw.parent_folder'),
             namespace_options={
-                'help': 'Inputs for the `PwBaseWorkChain` to run scf and nscf calculations.'})
-        # TODO optionally allow PwCalculation output parent_folder? # pylint: disable=fixme
-
-        # optional NSCF overrides
+                'help': 'Inputs for the `PwBaseWorkChain` to run `scf` and `nscf` calculations.',
+                'validator': validate_base_scf
+            }
+        )
         spec.expose_inputs(
-            PwBaseWorkChain, namespace='nscf',
-            include=('kpoints', 'kpoints_distance', 'kpoints_force_parity', 'pw.metadata.options'),
+            PwBaseWorkChain,
+            namespace='base_nscf',
+            exclude=('clean_workdir', 'pw.structure', 'pw.parent_folder'),
+            # include=('kpoints', 'kpoints_distance', 'kpoints_force_parity', 'pw.metadata.options', 'pw.parameters'),
             namespace_options={
-                'help': 'Optional inputs for nscf calculation, that override those set in the `base` namespace'})
-        spec.input('nscf.pw.metadata.options', valid_type=dict, required=False)
-
-        # DOS & PDOS
+                'help': 'Optional inputs for `nscf` calculation, that override those set in the `base` namespace.',
+                'validator': validate_base_nscf
+            }  # TODO optionally allow PwCalculation output parent_folder? # pylint: disable=fixme
```

Comment:
I guess the idea would be to allow the user to specify a parent folder for the `'nscf'` calculation, and skip the `'scf'`  one? (that's why I moved this todo here) In this case the `base_scf` input will become optional, and we can add validation at the top level to see that either the `base_scf` or `base_nscf.pw.parent_folder` inputs are provided.

Reply:
yep, obviously you may have already run a bunch of scf/relax calculations and don't want to have to repeat them to generate the PDOS

Reply:
I've updated the work chain to make the `base_scf` optional in dd34b5a.